### PR TITLE
[SBS-1][SBS-20] Sign Up

### PR DESCRIPTION
**Pull request by spartan-longnguyen**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

https://github.com/superbadteam/superbad-store_mobile/assets/143795927/5ca98601-f695-4bcf-ae8f-8e7c469b30c4

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/i18n/en.ts [hunk 1]
```diff
@@ -48,6 +49,18 @@ const en = {
...
+    signUp: "Sign up",
+    signUpWithEmail: "Sign up with Email",
+    nameFieldPlaceholder: "Full Name",
+    emailFieldPlaceholder: "Email Id",
+    passwordFieldPlaceholder: "Password",
+    passwordConfirmFieldPlaceholder: "Confirm Password",
+    nameFieldLabel: "Full Name",
+    emailFieldLabel: "Email",
+    passwordFieldLabel: "Password",
+    passwordConfirmFieldLabel: "Confirm Password",
```
Comment (phamhongphuc1403) [inline app/i18n/en.ts:62]: sao nhiều field trùng nhau v

File: app/screens/SignUpScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
+import { observer } from "mobx-react-lite";
+import React, { ComponentType, FC, useMemo, useRef, useState } from "react";
+import { TextInput, TextStyle, ViewStyle } from "react-native";
```
Comment (phamhongphuc1403) [inline app/screens/SignUpScreen.tsx:2]: dùng component có sẵn của boilerplate

File: app/i18n/en.ts (hunk 1 above)
Comment (spartan-longnguyen) [reply app/i18n/en.ts:62]: thì 1 cái label 1 cái text

File: app/screens/SignUpScreen.tsx (hunk 2 above)
Comment (spartan-longnguyen) [reply app/screens/SignUpScreen.tsx:2]: Tao cóp từ login qua mà ?

File: app/i18n/en.ts (hunk 1 above)
Comment (phamhongphuc1403) [reply app/i18n/en.ts:62]: xai chung di

Comment (spartan-longnguyen) [reply app/i18n/en.ts:62]: cái gì ra cái nấy

Comment (spartan-longnguyen) [reply app/i18n/en.ts:62]: xài như này sau có change text hay label thì sửa 1 chỗ thôi

**Comment by spartan-longnguyen** [comment]
LGTM

File: app/screens/SignUpScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  );
+
+  const handleSignUp = () => {
+    signUpByEmail(
+      { name, email: authEmail, password: authPassword, confirmPassword: authPasswordConfirm },
+      {
+        onSuccess: () => {
+          navigation.navigate("Login");
+        },
+      },
```
Comment (phamhongphuc1403) [inline app/screens/SignUpScreen.tsx]: bỏ vào useRegister đi m

Comment (spartan-longnguyen) [reply app/screens/SignUpScreen.tsx]: ừ

**Review by phamhongphuc1403 — APPROVED** [review]
LGTM

**Comment by huynamboz** [comment]
lgbt

**Review by huynamboz — APPROVED** [review]

**Pull request merged**